pkg/apiserver/registry/datapackaging/package_storage_client.go

```diff
@@ -152,7 +153,30 @@ func (t PackageTranslator) ToExternalWatcher(intObjWatcher watch.Interface, fiel
 }
 
 func (t PackageTranslator) ToExternalError(err error) error {
-	// TODO: implement
+
+	status, ok := err.(apierrors.APIStatus)
+	if !(ok || errors.As(err, &status)) {
+		return err
+	}
+
+	details := status.Status().Details
+	if details != nil && details.Kind == "internalpackages" && details.Group == internalpkgingv1alpha1.SchemeGroupVersion.Group {
+		packageName, translateErr := t.ToExternalName(details.Name)
+		if translateErr != nil {
+			return err
+		}
+
+		switch status.Status().Reason {
+		case metav1.StatusReasonNotFound:
+			return apierrors.NewNotFound(datapkgingv1alpha1.Resource("package"), packageName)
+		case metav1.StatusReasonAlreadyExists:
+			return apierrors.NewAlreadyExists(datapkgingv1alpha1.Resource("package"), packageName)
+		// TODO Handle other types of errors
```

Comment:
Do we plan on doing this for other errors? if not we can remove the TODO

Reply:
We need to plan on reproducing errors with these[ status reasons](https://github.com/kubernetes/apimachinery/blob/60eaa653342bd0cc48626fa2604bb6d3e7171706/pkg/api/errors/errors.go#L47) and handle the possible errors.

Reply:
If we are not planning to surface them separately in kapp-controller we should just remove the TODO since we are not planning on doing it

Reply:
Makes sense, I think we have covered the most common errors, we can think of handling other errors if there is a demand for it.